shieldlib/src/shield_renderer.ts

```diff
@@ -123,9 +131,25 @@ export class AbstractShieldRenderer {
   }
 
   /** Set which MaplibreGL map to handle shields for */
-  public renderOnMaplibreGL(map: Map): AbstractShieldRenderer {
+  public renderOnMaplibreGL(map: MapLibre): AbstractShieldRenderer {
     this.renderOnRepository(new MaplibreGLSpriteRepository(map));
     map.on("styleimagemissing", this.getStyleImageMissingHandler());
+    document.fonts.ready.then(() => {
+      this._fontsLoaded = true;
+      if (this._preFontImageCache.size == 0) {
+        return;
+      }
+      console.log("Re-processing shields with loaded fonts");
+
+      // Loop through each previously-loaded shield and re-render it
+      for (let [id, routeDef] of this._preFontImageCache.entries()) {
+        map.removeImage(id);
```

Comment:
Calling `removeImage()` and `addImage()` doesn’t seem to work. If I disable my browser’s cache, go directly to [Mechanisburg](https://preview.ourmap.us/pr/991/#map=14/40.07107/-83.5519), and zoom out and pan over the [Springfield](https://preview.ourmap.us/pr/991/#map=10/39.9396/-83.8312), I see the State Route 4 shield with the fallback font but the U.S. Route 68 and Interstate 70 shields with the downloaded Web font.

<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/243dcb1a-3fc6-4b19-9cf0-22cc15a4a110" width="400" alt="Springfield">

[`updateImage()`](https://maplibre.org/maplibre-gl-js/docs/API/classes/maplibregl.Map/#updateimage) is specifically designed for this purpose. Calling it would require refactoring `SpriteRepository.putSprite()`.

Reply:
Good catch.  I have a good reproducer as well.

Reply:
I'm actually not having expected results with updateImage() either:

![image](https://github.com/ZeLonewolf/openstreetmap-americana/assets/3254090/7878b09e-c992-47e1-b2b0-049b7923d208)

Is there something I can do to forcibly re-render the old shields beyond the sprite sheet update?

Reply:
As of maplibre/maplibre-gl-js#1805, the map is able to juggle multiple spritesheets (or “sprites” as that PR confusingly calls them). Could we add the shield sprites to one spritesheet, then dump that spritesheet once the fonts load? I wonder if that would clear the cache and convince the map to reevaluate the image expressions.